src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,123 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.ConnectionUtil;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String CREATION_EXCEPTION_MESSAGE = "Expected to insert 1 row,"
+                                                            + " but inserted 0 rows";
+    private static final String WRONG_ID_MESSAGE = "Cant find a book with id = ";
+    private static final String ID_COLUMN = "id";
+    private static final String TITLE_COLUMN = "title";
+    private static final String PRICE_COLUMN = "price";
+
+    private static final int MIN_AFFECTED_ROWS = 1;
+    private static final String CREATE_QUERY = "INSERT INTO books (title, price) VALUES(?, ?)";
+    private static final String FIND_BY_ID_QUERY = "SELECT * FROM books WHERE id = ?";
+    private static final String FIND_ALL_QUERY = "SELECT * FROM books";
+    private static final String UPDATE_QUERY = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+    private static final String DELETE_QUERY = "DELETE FROM books WHERE id = ?";
+
+    @Override
+    public Book create(Book book) {
+        String title = book.getTitle();
+        double price = book.getPrice().doubleValue();
+        try (PreparedStatement statement = ConnectionUtil.makeConnection()
+                .prepareStatement(CREATE_QUERY, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, title);
+            statement.setDouble(2, price);
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < MIN_AFFECTED_ROWS) {
+                throw new DataProcessingException(CREATION_EXCEPTION_MESSAGE);
+            }

```

Comment:
I see you perform checks on affected rows in multiple places. Try to provide some generic approach for that, in order to avoid code duplication